src/vault/auth/approle.clj

```diff
@@ -24,6 +24,84 @@
     mount instead of the default. Passing `nil` will reset the client to the
     default.")
 
+  (upsert-role!
+    [client role-name opts]
+    "Creates a new AppRole or updates an existing AppRole.
+
+    When creating or updating an AppRole, there must be at least
+    one option supplied
```

Comment:
Minor: several of the docstrings in this ns are missing periods at the end of sentences.